cae.py

```diff
@@ -208,14 +209,21 @@ def evaluate(self, ):
         self.printFinalResult(output, passed, numberOfTests)
 
     def setCompilerType(self):
-        if self.fileType == "jar":
+        if self.fileType == 'jar':
             self.compilerType = ['java', '-jar', self.pathToExecutable]
-        elif self.fileType == "py":
-            self.compilerType = ['python', self.pathToExecutable]
-        elif self.fileType == "rb":
-            self.compilerType = ['ruby', self.pathToExecutable]
-        elif self.fileType == "sh":
-            self.compilerType = ['bash', self.pathToExecutable]
+            return
+        
+        f = open(self.pathToExecutable, 'r')
+        firstLine = f.readline()
+        f.close()
+
+        if platform.system() == 'Windows' or firstLine[:3] != '#!/':
```

Comment:
What if Windows user is using Git bash? Also, Shebang interpreter doesn't have to start with `/`.